### PR TITLE
*: update the recent improvements about tls/security

### PR DESCRIPTION
**Pull request by lysu**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- https://github.com/pingcap/tidb/pull/14749: reload client tls via alter instance
- https://github.com/pingcap/tidb/pull/14833: reload cluster tls between tidb-tikv
- https://github.com/pingcap/tidb/pull/15071: reload cluster tls between tidb-pd
- https://github.com/pingcap/tidb/pull/14749: auth between cluster node via CN
- https://github.com/pingcap/tidb-ctl/pull/26: support tls betwen tidb-ctl and tidb-server

- This PR is translated from:
- Other reference link(s):

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/docs-cn/2417)

### CODE REVIEW

File: how-to/secure/enable-tls-clients.md [hunk 1]
```diff
@@ -7,21 +7,32 @@ category: how-to
...
-简单来说，要使用加密连接必须同时满足以下两个条件：
+TiDB 的加密连接支持默认是关闭的，必须在 TiDB 服务端通过配置开启加密连接的支持后，才能在客户端中使用加密连接，要使用加密连接必须同时满足以下两个条件：
 
 1. TiDB 服务端配置开启加密连接的支持
 2. 客户端指定使用加密连接
 
+另外，与 MySQL 一致，TiDB 加密连接是以单个连接为单位的，默认情况下是可选的，因而对于开启了加密连接支持的 TiDB 服务端，客户端既可以选择通过加密连接安全地连接到该 TiDB 服务端，也可以选择使用普通的非加密连接。如需强制要求客户端使用加密连接可以通过以下两种方式进行配置:
+
+1. 通过在启动参数中配置 `--require-secure-transport` 要求所有用户必须使用加密连接连接 TiDB
+2. 通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`) 时指定 `require ssl` 要求指定用户必须使用加密连接连接， TiDB 以创建用户为例:
```
Comment (lysu) [inline how-to/secure/enable-tls-clients.md:26]:
```suggestion
2. 通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`) 时指定 `require ssl` 要求指定用户必须使用加密连接连接 TiDB，以创建用户为例:
```

File: how-to/secure/enable-tls-clients.md [hunk 2]
```diff
@@ -7,21 +7,32 @@ category: how-to
...
+
+```sql
+create user 'u1'@'%'  require ssl;
+```
+
+> **注意：**
+>
+> 如果登录用户已配置使用[TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书信息时也会隐式要求对应用户必须使用加密连接连接 TiDB。
```
Comment (lysu) [inline how-to/secure/enable-tls-clients.md:34]:
```suggestion
> 如果登录用户已配置使用[TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书，也会隐式要求对应用户必须使用加密连接连接 TiDB。
```

File: how-to/secure/enable-tls-clients.md [hunk 3]
```diff
@@ -84,9 +101,15 @@ MySQL 5.7 及以上版本自带的客户端默认尝试使用安全连接，若
...
+默认情况，服务端对客户端的身份验证是可选的。若客户端在 TLS 握手时未出示自己的身份证书，也能正常建立 TLS 连接。但也可以通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`)时指定 `require 509` 要求客户端必须提供身份证明， TiDB 以创建用户为例:
+
+```sql
+create user 'u1'@'%'  require x509;
+```
+
 > **注意：**
 >
-> 目前 TiDB 尚不支持强制验证客户端身份，即服务端对客户端的身份验证是可选的。若客户端在 TLS 握手时未出示自己的身份证书，也能正常建立 TLS 连接。
+> 如果登录用户已配置使用[TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书信息时也会隐式要求对应用户必须使用加密连接连接 TiDB
```
Comment (lysu) [inline how-to/secure/enable-tls-clients.md:112]:
```suggestion
> 如果登录用户已配置使用[TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书，也会隐式要求对应用户需进行身份验证
```

File: how-to/secure/enable-tls-clients.md [hunk 4]
```diff
@@ -84,9 +101,15 @@ MySQL 5.7 及以上版本自带的客户端默认尝试使用安全连接，若
 - 若要使 TiDB 服务端验证 MySQL 客户端身份，TiDB 服务端需配置 `ssl-cert`、`ssl-key`、`ssl-ca` 参数，客户端需至少指定 `--ssl-cert`、`--ssl-key` 参数。必须确保服务端配置的证书和客户端配置的证书都是由服务端配置指定的 `ssl-ca` 签发的。
 - 若要进行双向身份验证，请同时满足上述要求。
 
+默认情况，服务端对客户端的身份验证是可选的。若客户端在 TLS 握手时未出示自己的身份证书，也能正常建立 TLS 连接。但也可以通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`)时指定 `require 509` 要求客户端必须提供身份证明， TiDB 以创建用户为例:
```
Comment (lysu) [inline how-to/secure/enable-tls-clients.md:104]:
```suggestion
默认情况，服务端对客户端的身份验证是可选的。若客户端在 TLS 握手时未出示自己的身份证书，也能正常建立 TLS 连接。但也可以通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`) 时指定 `require 509` 要求客户端需进行身份验证， 以创建用户为例:
```

File: reference/sql/statements/alter-instance.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+`ALTER INSTANCE` 语句用于对单个 TiDB 实例进行变更操作， 目前 TiDB 仅支持 `RELOAD TLS` 子句。
+
+# RELOAD TLS
+
+`ALTER INSTACE RELOAD TLS` 语句用于从原配置的证书([`ssl-cert`](/reference/configuration/tidb-server/configuration-file.md#ssl-cert)), 密钥([`ssl-key`](/reference/configuration/tidb-server/configuration-file.md#ssl-key)) 和 CA([`ssl-ca`](/reference/configuration/tidb-server/configuration-file.md#ssl-ca))路径重新加证书，密钥和 CA。
+
+新加载的证书密钥和 CA 将在语句执行成功后对新建立的连接生效，不会影响语句执行前已建立的连接。
+
+在重加载遇到错误时默认会报错且继续使用变更前的密钥和证书，而在添加可选的 `NO ROLLBACK ON ERROR` 后重加载遇到错误将不报错并关闭 TLS 安全连接功能继续工作。
```
Comment (lysu) [inline reference/sql/statements/alter-instance.md:17]:
```suggestion
在重加载遇到错误时默认会报错返回且继续使用变更前的密钥和证书，但在添加可选的 `NO ROLLBACK ON ERROR` 后遇到错误将不报错并以关闭 TLS 安全连接功能的方式继续工作。
```

**Comment by lysu** [comment]
@TomShawn it's ready to review, PTAL thx~

File: how-to/secure/enable-tls-clients.md [hunk 6]
```diff
@@ -64,6 +75,12 @@ ssl-key = "certs/server-key.pem"
 
 若证书参数无误，则 TiDB 在启动时将会输出 `secure connection is enabled`，否则 TiDB 会输出 `secure connection is NOT ENABLED`。
 
+## 重加载证书密钥和 CA
+
+在需要替换证书，密钥或 CA 时，可以在完成对应文件替换后，对运行中的 TiDB 实例 执行 [`ALTER INSTACE RELOAD TLS`](/reference/sql/statements/alter-instance.md) 语句从原配置的证书([`ssl-cert`](/reference/configuration/tidb-server/configuration-file.md#ssl-cert)), 密钥([`ssl-key`](/reference/configuration/tidb-server/configuration-file.md#ssl-key)) 和 CA([`ssl-ca`](/reference/configuration/tidb-server/configuration-file.md#ssl-ca)) 的路径重新加证书，密钥和 CA 文件, 而无需重启 TiDB 实例.
```
Comment (TomShawn) [inline how-to/secure/enable-tls-clients.md:80]:
```suggestion
在需要替换证书、密钥或 CA 时，可以在完成对应文件替换后，对运行中的 TiDB 实例执行 [`ALTER INSTACE RELOAD TLS`](/reference/sql/statements/alter-instance.md) 语句从原配置的证书 ([`ssl-cert`](/reference/configuration/tidb-server/configuration-file.md#ssl-cert))、密钥 ([`ssl-key`](/reference/configuration/tidb-server/configuration-file.md#ssl-key)) 和 CA ([`ssl-ca`](/reference/configuration/tidb-server/configuration-file.md#ssl-ca)) 的路径重新加证书、密钥和 CA 文件，而无需重启 TiDB 实例。
```

File: how-to/secure/enable-tls-clients.md [hunk 7]
```diff
@@ -84,9 +101,15 @@ MySQL 5.7 及以上版本自带的客户端默认尝试使用安全连接，若
 - 若要使 TiDB 服务端验证 MySQL 客户端身份，TiDB 服务端需配置 `ssl-cert`、`ssl-key`、`ssl-ca` 参数，客户端需至少指定 `--ssl-cert`、`--ssl-key` 参数。必须确保服务端配置的证书和客户端配置的证书都是由服务端配置指定的 `ssl-ca` 签发的。
 - 若要进行双向身份验证，请同时满足上述要求。
 
+默认情况，服务端对客户端的身份验证是可选的。若客户端在 TLS 握手时未出示自己的身份证书，也能正常建立 TLS 连接。但也可以通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`) 时指定 `require 509` 要求客户端需进行身份验证， 以创建用户为例:
```
Comment (TomShawn) [inline how-to/secure/enable-tls-clients.md:104]:
```suggestion
默认情况，服务端对客户端的身份验证是可选的。若客户端在 TLS 握手时未出示自己的身份证书，也能正常建立 TLS 连接。但也可以通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`) 时指定 `require 509` 要求客户端需进行身份验证，以创建用户为例：
```

File: how-to/secure/enable-tls-clients.md [hunk 8]
```diff
@@ -84,9 +101,15 @@ MySQL 5.7 及以上版本自带的客户端默认尝试使用安全连接，若
...
+
+```sql
+create user 'u1'@'%'  require x509;
+```
+
 > **注意：**
 >
-> 目前 TiDB 尚不支持强制验证客户端身份，即服务端对客户端的身份验证是可选的。若客户端在 TLS 握手时未出示自己的身份证书，也能正常建立 TLS 连接。
+> 如果登录用户已配置使用[TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书，也会隐式要求对应用户需进行身份验证
```
Comment (TomShawn) [inline how-to/secure/enable-tls-clients.md:112]:
```suggestion
> 如果登录用户已配置使用 [TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书，也会隐式要求对应用户需进行身份验证。
```

File: how-to/secure/enable-tls-clients.md [hunk 9]
```diff
@@ -7,21 +7,32 @@ category: how-to
...
+1. 通过在启动参数中配置 `--require-secure-transport` 要求所有用户必须使用加密连接连接 TiDB
+2. 通过在创建用户 (`create user`)，赋予权限 (`grant`) 或修改已有用户 (`alter user`) 时指定 `require ssl` 要求指定用户必须使用加密连接连接 TiDB，以创建用户为例:
+
+```sql
+create user 'u1'@'%'  require ssl;
+```
+
+> **注意：**
+>
+> 如果登录用户已配置使用[TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书，也会隐式要求对应用户必须使用加密连接连接 TiDB。
```
Comment (TomShawn) [inline how-to/secure/enable-tls-clients.md:34]:
```suggestion
> 如果登录用户已配置使用 [TiDB 证书鉴权功能](/reference/security/cert-based-authentication.md#配置登陆时需要校验的用户证书信息)校验用户证书，也会隐式要求对应用户必须使用加密连接连接 TiDB。
```

File: reference/sql/statements/alter-instance.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
+---
+title: ALTER INSTANCE
+summary: TiDB 数据库中 ALTER INSTANCE 的使用概况。
+category: reference
+---
+
+# ALTER INSTANCE
+
+`ALTER INSTANCE` 语句用于对单个 TiDB 实例进行变更操作， 目前 TiDB 仅支持 `RELOAD TLS` 子句。
```
Comment (TomShawn) [inline reference/sql/statements/alter-instance.md:9]:
```suggestion
`ALTER INSTANCE` 语句用于对单个 TiDB 实例进行变更操作。目前 TiDB 仅支持 `RELOAD TLS` 子句。
```

File: reference/sql/statements/alter-instance.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+# RELOAD TLS
```
Comment (TomShawn) [inline reference/sql/statements/alter-instance.md:11]:
```suggestion
## RELOAD TLS
```

File: reference/sql/statements/alter-instance.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+`ALTER INSTACE RELOAD TLS` 语句用于从原配置的证书([`ssl-cert`](/reference/configuration/tidb-server/configuration-file.md#ssl-cert)), 密钥([`ssl-key`](/reference/configuration/tidb-server/configuration-file.md#ssl-key)) 和 CA([`ssl-ca`](/reference/configuration/tidb-server/configuration-file.md#ssl-ca))路径重新加证书，密钥和 CA。
```
Comment (TomShawn) [inline reference/sql/statements/alter-instance.md:13]:
```suggestion
`ALTER INSTACE RELOAD TLS` 语句用于从原配置的证书 ([`ssl-cert`](/reference/configuration/tidb-server/configuration-file.md#ssl-cert))、密钥 ([`ssl-key`](/reference/configuration/tidb-server/configuration-file.md#ssl-key)) 和 CA ([`ssl-ca`](/reference/configuration/tidb-server/configuration-file.md#ssl-ca)) 的路径重新加证书、密钥和 CA。
```

File: reference/sql/statements/alter-instance.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+新加载的证书密钥和 CA 将在语句执行成功后对新建立的连接生效，不会影响语句执行前已建立的连接。
+
+在重加载遇到错误时默认会报错返回且继续使用变更前的密钥和证书，但在添加可选的 `NO ROLLBACK ON ERROR` 后遇到错误将不报错并以关闭 TLS 安全连接功能的方式继续工作。
```
Comment (TomShawn) [inline reference/sql/statements/alter-instance.md:17]: 是什么`继续工作`呢？

File: reference/sql/statements/alter-instance.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER INSTANCE RELOAD TLS;
+```
+
+## MySQL 兼容性
+
+- 仅支持从原配置路径重加载， 不支持动态修改加载路径，也不支持动态启用启动时未开启的 TLS 加密连接功能。
```
Comment (TomShawn) [inline reference/sql/statements/alter-instance.md:33]:
```suggestion
仅支持从原配置路径重加载，不支持动态修改加载路径，也不支持动态启用启动 TiDB 时未开启的 TLS 加密连接功能。
```

File: reference/sql/statements/alter-instance.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+
+`ALTER INSTANCE` 语句用于对单个 TiDB 实例进行变更操作。目前 TiDB 仅支持 `RELOAD TLS` 子句。
+
+## RELOAD TLS
+
+`ALTER INSTACE RELOAD TLS` 语句用于从原配置的证书 ([`ssl-cert`](/reference/configuration/tidb-server/configuration-file.md#ssl-cert))、密钥 ([`ssl-key`](/reference/configuration/tidb-server/configuration-file.md#ssl-key)) 和 CA ([`ssl-ca`](/reference/configuration/tidb-server/configuration-file.md#ssl-ca)) 的路径重新加证书、密钥和 CA。
+
+新加载的证书密钥和 CA 将在语句执行成功后对新建立的连接生效，不会影响语句执行前已建立的连接。
+
+在重加载遇到错误时默认会报错返回且继续使用变更前的密钥和证书，但在添加可选的 `NO ROLLBACK ON ERROR` 后遇到错误将不报错并以关闭 TLS 安全连接功能的方式继续工作。
```
Comment (lysu) [inline reference/sql/statements/alter-instance.md:17]:
```suggestion
在重加载遇到错误时默认会报错返回且继续使用变更前的密钥和证书，但在添加可选的 `NO ROLLBACK ON ERROR` 后遇到错误将不报错并以关闭 TLS 安全连接功能的方式处理后续请请求。
```

File: how-to/secure/enable-tls-between-components.md [hunk 16]
```diff
@@ -86,6 +92,47 @@ MySQL Client 与 TiDB 之间使用一套证书，TiDB 集群组件之间使用
...
+
+        在 `config` 文件或命令行参数中设置：
+
+        ```toml
+        [security]
+        cluster-verify-cn = [
+          "TiDB-Server",
+          "TiKV-Control",          
+        ]
+
```
Comment (lysu) [inline how-to/secure/enable-tls-between-components.md:109]:
```suggestion
        ]
```

File: how-to/secure/enable-tls-between-components.md [hunk 17]
```diff
@@ -86,6 +92,47 @@ MySQL Client 与 TiDB 之间使用一套证书，TiDB 集群组件之间使用
     ./tikv-ctl --host="127.0.0.1:20160" --ca-path="/path/to/ca.pem" --cert-path="/path/to/client.pem" --key-path="/path/to/clinet-key.pem"
     ```
 
+3. 配置校验调用者 Common Name。
+
+    通常被调用者除了校验调用者提供的密钥、证书和 CA 有效性外，还需要校验调用方身份(比如: 只能 TiKV 只能被 TiDB 访问，需阻止拥有合法证书但并非 TiDB 的其他访问者访问 TiKV)。推荐在生成证书时通过 `Common Name` 标识证书使用者身份，并在被调用者配置检查证书 `Common Name` 列表来检查调用者身份。
+
+    - TiDB
+
+        在 `config` 文件或命令行参数中设置：
```
[the remark this answers is not in the document]
Comment (lysu) [reply how-to/secure/enable-tls-between-components.md:101]: 是的， 和前面一段 https://github.com/pingcap/docs-cn/blob/c1a2f7df408b3a2d88287d6cc72c6ec892070960/how-to/secure/enable-tls-between-components.md#tidb-%E9%9B%86%E7%BE%A4%E7%BB%84%E4%BB%B6%E9%97%B4%E5%BC%80%E5%90%AF-tls%E5%8F%8C%E5%90%91%E8%AE%A4%E8%AF%81 一样在改配置

File: how-to/secure/enable-tls-between-components.md (hunk 16 above)
Comment (lysu) [inline how-to/secure/enable-tls-between-components.md:109]:
```suggestion
```

File: reference/sql/statements/alter-instance.md [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+
+`ALTER INSTANCE` 语句用于对单个 TiDB 实例进行变更操作。目前 TiDB 仅支持 `RELOAD TLS` 子句。
+
+## RELOAD TLS
+
+`ALTER INSTACE RELOAD TLS` 语句用于从原配置的证书 ([`ssl-cert`](/reference/configuration/tidb-server/configuration-file.md#ssl-cert))、密钥 ([`ssl-key`](/reference/configuration/tidb-server/configuration-file.md#ssl-key)) 和 CA ([`ssl-ca`](/reference/configuration/tidb-server/configuration-file.md#ssl-ca)) 的路径重新加证书、密钥和 CA。
+
+新加载的证书密钥和 CA 将在语句执行成功后对新建立的连接生效，不会影响语句执行前已建立的连接。
+
+在重加载遇到错误时默认会报错返回且继续使用变更前的密钥和证书，但在添加可选的 `NO ROLLBACK ON ERROR` 后遇到错误将不报错并以关闭 TLS 安全连接功能的方式处理后续请请求。
```
Comment (lysu) [inline reference/sql/statements/alter-instance.md:17]:
```suggestion
在重加载遇到错误时默认会报错返回且继续使用变更前的密钥和证书，但在添加可选的 `NO ROLLBACK ON ERROR` 后遇到错误将不报错并以关闭 TLS 安全连接功能的方式处理后续请求。
```

**Pull request merged**